tuf/api/metadata.py

```diff
@@ -183,7 +194,14 @@ def from_bytes(
 
             deserializer = JSONDeserializer()
 
-        return deserializer.deserialize(data)
+        md = deserializer.deserialize(data)
+
+        # Ensure deserialized signed type matches the requested type
+        if signed_type is not None and signed_type != type(md.signed):
```

Comment:
Is `Generic` needed for the `signed_type` check to work or they serve two different purposes (independent of each other)?
- making Metadata class Generic over T for type checking
- signed_type argument for runtime check

Reply:
the runtime check certainly works without any generics, yes.

They are still related in that providing `signed_type` also defines what the specific type T is: if the signed_type argument is not used, some other means needs to be used to get a specific type for static typing (like the syntax `md = Metadata[Snapshot].from_bytes(data)`)